spec/aruba/api/filesystem_spec.rb

```diff
@@ -0,0 +1,766 @@
+          end
+        end
+      end
+
+      context 'when source is non-existing' do
+        it { expect { @aruba.copy source, destination }.to raise_error ArgumentError }
+      end
+    end
+  end
+
+  context '#write_file' do
+    it 'writes file' do
+      @aruba.write_file(@file_name, '')
+
+      expect(File.exist?(@file_path)).to eq true
+    end
+  end
+
+  context '#write_fixed_size_file' do
+    it "should write a fixed sized file" do
+      @aruba.write_fixed_size_file(@file_name, @file_size)
+      expect(File.exist?(@file_path)).to eq true
+      expect(File.size(@file_path)).to eq @file_size
+    end
+
+    it "works with ~ in path name" do
+      file_path = File.join('~', random_string)
+
+      @aruba.with_environment 'HOME' => File.expand_path(aruba.current_directory) do
+        @aruba.write_fixed_size_file(file_path, @file_size)
+
+        expect(File.exist?(File.expand_path(file_path))).to eq true
+        expect(File.size(File.expand_path(file_path))).to eq @file_size
+      end
+    end
+  end
+
+  describe '#chmod' do
+    def actual_permissions
+      format("%o" , File::Stat.new(file_path).mode )[-4,4]
+    end
+
+    let(:file_name) { @file_name }
+    let(:file_path) { @file_path }
+    let(:permissions) { '0655' }
+
+    before :each do
+      @aruba.set_environment_variable 'HOME', File.expand_path(@aruba.aruba.current_directory)
+    end
+
+    before(:each) do
+      File.open(file_path, 'w') { |f| f << "" }
+    end
+
+    before(:each) do
+      @aruba.chmod(permissions, file_name)
+    end
+
+    context 'when file exists' do
+      context 'and permissions are given as string' do
+        it { expect(actual_permissions).to eq('0655') }
+      end
+
+      context 'and permissions are given as octal number' do
+        let(:permissions) { 0o655 }
+        it { expect(actual_permissions).to eq('0655') }
+      end
+
+      context 'and path has ~ in it' do
+        let(:path) { random_string }
+        let(:file_name) { File.join('~', path) }
+        let(:file_path) { File.join(@aruba.aruba.current_directory, path) }
+
+        it { expect(actual_permissions).to eq('0655') }
+      end
+    end
+  end
+
+  context "#with_file_content" do
+    before :each do
+      @aruba.write_file(@file_name, "foo bar baz")
+    end
+
+    it "checks the given file's full content against the expectations in the passed block" do
+      @aruba.with_file_content @file_name do |full_content|
+        expect(full_content).to eq "foo bar baz"
+      end
+    end
+
+    it "works with ~ in path name" do
+      file_path = File.join('~', random_string)
+
+      @aruba.with_environment 'HOME' => File.expand_path(aruba.current_directory) do
+        @aruba.write_file(file_path, "foo bar baz")
+
+        @aruba.with_file_content file_path do |full_content|
+          expect(full_content).to eq "foo bar baz"
+        end
+      end
+    end
+
+    context "checking the file's content against the expectations in the block" do
```

Comment:
probably should elude to regex here in this context

Reply:
These tests were moved from another file, but I don't think they add anything so I'm going to remove them.